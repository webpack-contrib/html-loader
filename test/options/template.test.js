/* eslint-disable
  prefer-destructuring,
*/
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('template', () => {
    test('{Boolean}', async () => {
      const config = {
        loader: {
          test: /\.html$/,
          options: {
            template: true,
          },
        },
      };

      const stats = await webpack('options/template/fixture.js', config);
      const { source } = stats.toJson().modules[1];

      expect(source).toMatchSnapshot();
    });

    test('{String}', async () => {
      const config = {
        loader: {
          test: /\.html$/,
          options: {
            template: '$',
          },
        },
      };

      const stats = await webpack('options/template/fixture.js', config);
      const { source } = stats.toJson().modules[1];

      expect(source).toMatchSnapshot();
    });
  });
});
