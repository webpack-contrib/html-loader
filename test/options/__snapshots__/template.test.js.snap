// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options template {Boolean} 1`] = `
"// HTML Imports
import HTML__URL__0 from './urls/file.png';
import HTML__IMPORT__0 from './imports/import.html';


// HTML Exports


// HTML
export default function (_) { return \`<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"utf-8\\">
  <title>HTML Loader</title>
</head>
<body>
  <!-- HTML Template -->
  <div id=\\"app\\">\${ _.hello }</div>
  <!-- HTML Import -->
  \${HTML__IMPORT__0(_)}
  <!-- HTML URL -->
  <img src=\\"\${HTML__URL__0}\\">
</body>
</html>
\`; }"
`;

exports[`Options template {String} 1`] = `
"// HTML Imports
import HTML__URL__0 from './urls/file.png';
import HTML__IMPORT__0 from './imports/import.html';


// HTML Exports


// HTML
export default function ($) { return \`<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"utf-8\\">
  <title>HTML Loader</title>
</head>
<body>
  <!-- HTML Template -->
  <div id=\\"app\\">\${ _.hello }</div>
  <!-- HTML Import -->
  \${HTML__IMPORT__0($)}
  <!-- HTML URL -->
  <img src=\\"\${HTML__URL__0}\\">
</body>
</html>
\`; }"
`;
